test/e2e/app-dir/use-cache/use-cache.test.ts

```diff
@@ -25,4 +28,41 @@ describe('use-cache', () => {
     // The navigation to n=2 should be some other random value.
     expect(random1a).not.toBe(random2)
   })
+
+  it('should dedupe with react cache inside "use cache"', async () => {
+    const browser = await next.browser('/react-cache')
+    const a = await browser.waitForElementByCss('#a').text()
+    const b = await browser.waitForElementByCss('#b').text()
+    // TODO: This is broken. It is expected to pass once we fix it.
+    expect(a).not.toBe(b)
+  })
+
+  it('should error when cookies/headers/draftMode is used inside "use cache"', async () => {
+    const browser = await next.browser('/errors')
+    expect(await browser.waitForElementByCss('#cookies').text()).toContain(
+      isNextDev
+        ? '`cookies` cannot be called inside "use cache".'
+        : GENERIC_RSC_ERROR
+    )
+    expect(await browser.waitForElementByCss('#headers').text()).toContain(
+      isNextDev
+        ? '`headers` cannot be called inside "use cache".'
+        : GENERIC_RSC_ERROR
+    )
+    expect(await browser.waitForElementByCss('#draft-mode').text()).toContain(
+      isNextDev
+        ? '`draftMode` cannot be called inside "use cache".'
+        : GENERIC_RSC_ERROR
+    )
+
+    expect(next.cliOutput).toContain(
+      '`cookies` cannot be called inside "use cache".'
+    )
+    expect(next.cliOutput).toContain(
+      '`headers` cannot be called inside "use cache".'
+    )
+    expect(next.cliOutput).toContain(
+      '`draftMode` cannot be called inside "use cache".'
+    )
```

Comment:
There's currently a quirk with `next.cliOutput` where it'll fail on deployment tests because `next.cliOutput` doesn't currently grab runtime logs from Vercel, only build logs.

It's not failing on your PR because we aren't able to run deployment tests from forks but will fail once on canary. We'll want to gate anything that depends on runtime logs to not run on `isNextDeploy` from `nextTestSetup`

Reply:
I gated it on `isNextDeploy`.